src/common/SearchEmptyState.js

```diff
@@ -0,0 +1,55 @@
+/* @flow */
+import React from 'react';
+import { StyleSheet, View } from 'react-native';
+import { ZulipButton, Label } from './';
+
+const styles = StyleSheet.create({
+  emptyState: {
+    flex: 1
+  },
+  container: {
+    alignSelf: 'stretch',
+    padding: 20
+  },
+  text: {
+    fontSize: 18,
+    textAlign: 'center'
+  },
+  button: {
+    marginTop: 15,
+    fontSize: 12
+  }
+});
+
+export default class SearchEmptyState extends React.PureComponent {
+  props: {
+    text?: string,
+    buttonText?: string,
+    buttonAction?: () => void,
+  };
+
+  static defaultProps = {
+    text: 'No Results',
+    buttonText: 'Show All'
+  };
+
+  render() {
+    const { text, buttonText, buttonAction } = this.props;
+
+    return (
+      <View style={styles.emptyState}>
+        <View style={styles.container}>
+          <View>
```

Comment:
Yeah, I am pretty sure you can do it with a single View even, not three. Use the flexbox css properties.